test/e2e/azure_test.go

```diff
@@ -499,7 +499,7 @@ var _ = Describe("Workload cluster creation", func() {
 					ClusterctlConfigPath:     clusterctlConfigPath,
 					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
 					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
-					Flavor:                   "aks",
+					Flavor:                   aksClusterNameSuffix,
```

Comment:
could be risky to reuse the same variable here as the flavor name doesn't necessarily need to match the cluster name suffix, it just happens to be true right now because we're using "aks" for both but we could decide to rename the template in the future and this would no longer work.